lib/charms/kubernetes_charm_libraries/v0/volumes.py

```diff
@@ -0,0 +1,415 @@
+        container = Container(
+            name=container_name,
+            volumeMounts=[requested_volume.volume_mount for requested_volume in requested_volumes],
+            resources=ResourceRequirements(limits={}, requests={}),
+        )
+        if any(
+            [
+                requested_volume.volume.emptyDir.medium == "HugePages"
+                for requested_volume in requested_volumes
+            ]
+        ):
+            container.resources.limits.update({"hugepages-1Gi": "2Gi"})
+            container.resources.requests.update({"hugepages-1Gi": "2Gi"})
+        statefulset_delta = StatefulSet(
+            spec=StatefulSetSpec(
+                selector=statefulset.spec.selector,  # type: ignore[attr-defined]
+                serviceName=statefulset.spec.serviceName,  # type: ignore[attr-defined]
+                template=PodTemplateSpec(
+                    spec=PodSpec(
+                        containers=[container],
+                        volumes=[
+                            requested_volume.volume for requested_volume in requested_volumes
+                        ],
+                    ),
+                ),
+            )
+        )
+        try:
+            self.client.patch(
+                res=StatefulSet,
+                name=statefulset_name,
+                obj=statefulset_delta,
+                patch_type=PatchType.APPLY,
+                namespace=self.namespace,
+                field_manager=self.__class__.__name__,
+            )
+        except ApiError:
+            raise KubernetesRequestedVolumesError(
+                f"Could not patch statefulset {statefulset_name}"
+            )
+        logger.info("Requested volumes added to %s statefulset", statefulset_name)
+
+    def remove_volumes(
+        self,
+        statefulset_name: str,
+        requested_volumes: Iterable[RequestedVolume],
+        container_name: str,
+    ) -> None:
+        """Replaces a statefulset removing requested volumes.
+
+        Args:
+            statefulset_name: Statefulset name
+            requested_volumes: Iterable of requested volumes
+            container_name: Container name
+        """
+        if not requested_volumes:
+            logger.info("No requested volumes were provided")
+            return
+        try:
+            statefulset = self.client.get(
+                res=StatefulSet, name=statefulset_name, namespace=self.namespace
+            )
+        except ApiError:
+            raise KubernetesRequestedVolumesError(f"Could not get statefulset {statefulset_name}")
+        containers: Iterable[Container] = statefulset.spec.template.spec.containers  # type: ignore[attr-defined]  # noqa: E501
+        requested_volumes_mounts = [
+            requested_volume.volume_mount for requested_volume in requested_volumes
+        ]
+        container = next(container for container in containers if container.name == container_name)
```

Comment:
If the required container could not be found, it will raise StopIteration error.